src/transmit.cpp

```diff
@@ -1236,6 +1303,21 @@ void tx_send_packets(struct tx *tx, struct rtp *rtpSession, const std::vector<in
             }
         }
 
+        // If packet pacing is enabled, then we should compare the actual
+        // performance against our expected performance. If we are ahead
+        // we should wait until we're at our expected timing.
+        if(packetPace) {
+            // Get the time now
+            std::chrono::high_resolution_clock::time_point now = std::chrono::high_resolution_clock::now();
+            // Calculate where we should be according to our target
+            std::chrono::duration targetDuration = std::chrono::nanoseconds(packetDurationTarget) * (loopIndex + 1);
+            std::chrono::high_resolution_clock::time_point target = start + targetDuration;
+            // Loop until we're on target
+            while(now < target) {
+                now = std::chrono::high_resolution_clock::now();
```

Comment:
Would it be worth putting a `std::this_thread::sleep_for` call into this loop?

Reply:
As discussed in slack:
`sleep_for` appears to always sleep for 50 microseconds even if you give a lower sleep timer - I would imagine that `sleep_until` have similar behaviour. I think that this would introduce potentially weird behaviour where you'd see groupings of packets. If we're sending, as an example 1200 packets and they need to be sent within 28ms. Then the average timing between packets is 23μs.